tests/modules/hooks.c

```diff
@@ -287,9 +289,51 @@ void keyInfoCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub, void
     }
 
     RedisModuleKeyInfoV1 *ei = data;
-    RedisModuleString *keyname = RedisModule_CreateStringPrintf(ctx, "key-info-%s", ei->key);
-    LogStringEvent(ctx, RedisModule_StringPtrLen(keyname, NULL), ei->key);
-    RedisModule_FreeString(ctx, keyname);
+    RedisModuleString *key = ei->key;
+    const char* keyname = RedisModule_StringPtrLen(key, NULL);
+    RedisModuleString *event_keyname = RedisModule_CreateStringPrintf(ctx, "key-info-%s", keyname);
+    LogStringEvent(ctx, RedisModule_StringPtrLen(event_keyname, NULL), keyname);
+    RedisModule_FreeString(ctx, event_keyname);
+
+    /* Open Key to check key exists */
+    RedisModuleKey *kp = RedisModule_OpenKey(ctx, key, REDISMODULE_READ);
+    if(kp != NULL) {
+        int nokey;
+        RedisModule_DictGetC(removed_event_log, (void*)keyname, strlen(keyname), &nokey);
+        if(nokey){
+            RedisModuleString *v = RedisModule_HoldString(ctx, key);
+            /* For string type, we keep value instead of key */
+            if (RedisModule_KeyType(kp) == REDISMODULE_KEYTYPE_STRING) {
```

Comment:
do we wanna handle other types as well? i.e. read the data? i'm not sure it matters.
but i do think we wanted to make sure we can read the expiry information!

Reply:
For other types, it is empty when we receive event like `LPOP`.